watertap/property_models/NDMA_prop_pack.py

```diff
@@ -753,27 +761,16 @@ def calculate_scaling_factors(self):
             )  # temperature change on the order of 1e1
             iscale.set_scaling_factor(self.enth_flow, sf)
 
-        # property relationships with phase index, but simple constraint
-        v_str_lst_simple = ["dens_mass_phase", "flow_vol_phase"]
-        for v_str in v_str_lst_simple:
-            if self.is_property_constructed(v_str):
-                v = getattr(self, v_str)
-                sf = iscale.get_scaling_factor(v, default=1, warning=True)
-                c = getattr(self, "eq_" + v_str)
-                iscale.constraint_scaling_transform(c, sf)
-
-        # property relationships indexed by component and phase
-        for v_str in (
-            "mass_frac_phase_comp",
-            "conc_mass_phase_comp",
-            "flow_mol_phase_comp",
-            "mole_frac_phase_comp",
-            "molality_phase_comp",
-        ):
-            if self.is_property_constructed(v_str):
-                v_comp = self.component(v_str)
-                for j, c in self.component("eq_" + v_str).items():
-                    sf = iscale.get_scaling_factor(
-                        v_comp["Liq", j], default=1, warning=True
-                    )
+        # transforming constraints
+        for metadata_dic in self.params.get_metadata().properties.values():
+            var_str = metadata_dic["name"]
+            if metadata_dic["method"] is not None and self.is_property_constructed(
+                var_str
+            ):
+                var = getattr(self, var_str)
+                if isinstance(var, Expression):
+                    continue  # properties that are expressions do not have constraints
+                con = getattr(self, "eq_" + var_str)
+                for ind, c in con.items():
+                    sf = iscale.get_scaling_factor(var[ind], default=1, warning=True)
                     iscale.constraint_scaling_transform(c, sf)
```

Comment:
I believe this can become a utility function: `transform_property_constraints`, in which all of the property packages that you modified in this PR can use

This could be located in `watertap/core/util` with a new file called `scaling.py`

Reply:
I assume the scaling.py file is small enough that it does not need a test file?